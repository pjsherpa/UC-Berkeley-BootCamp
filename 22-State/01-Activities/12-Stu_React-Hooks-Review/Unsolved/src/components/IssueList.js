import React from 'react';
import IssueItem from '../components/IssueItem';

// Here we accept an array of issues as a prop
const IssueList = ({ issues }) => {
  console.log('IssueList -> issues', issues);

  // We create a list of rendered IssueItems by using the map method on `issues`
  const renderedList = issues.map((issue) => {
    return <IssueItem key={issue.id} issue={issue} />;
  });

  // Here we return our array of IssueItems wrapped inside a parent div
  return <div className="ui relaxed divided list">{renderedList}</div>;
};

export default IssueList;
