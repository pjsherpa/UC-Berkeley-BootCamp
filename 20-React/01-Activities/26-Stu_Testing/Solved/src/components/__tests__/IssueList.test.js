import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import IssueList from '../IssueList';

let container = null;

beforeEach(() => {
  // Setup a DOM element as the target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // Cleanup on exiting to prevent this test from altering the results of future tests
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const issues = [
  {
    url: 'https://api.github.com/repos/microsoft/vscode/issues/68',
    html_url: 'https://github.com/microsoft/vscode/issues/68',
    id: 117635943,
    node_id: 'MDU6SXNzdWUxMTc2MzU5NDM=',
    number: 68,
    title: 'Git: Support git history in VSCode',
  },
  {
    url: 'https://api.github.com/repos/microsoft/vscode/issues/127',
    html_url: 'https://github.com/microsoft/vscode/issues/127',
    id: 117711073,
    node_id: 'MDU6SXNzdWUxMTc3MTEwNzM=',
    number: 127,
    title: 'Provide option to opt out of line ending normalisation for files',
  },
];

describe('IssueList', () => {
  // In this example, we render the IssueList component and format the rendered HTML with the pretty package before saving it as an snapshot.
  it('should render', () => {
    // Render the component inside the target container
    act(() => {
      render(<IssueList issues={issues} />, container);
    });

    // Format the rendered HTML with the pretty package
    const html = pretty(container.innerHTML);

    // Save the rendered HTML as an snapshot
    expect(html).toMatchSnapshot();
  });

  // In this example, we check to see if the issueList contains the text "Git: Support git history in VSCode"
  it('should contain text', () => {
    // Render the component
    act(() => {
      render(<IssueList issues={issues} />, container);
    });

    // Check to see if the rendered HTML contains the text "Git: Support git history in VSCode"
    expect(container.innerHTML).toContain('Git: Support git history in VSCode');
  });
});
