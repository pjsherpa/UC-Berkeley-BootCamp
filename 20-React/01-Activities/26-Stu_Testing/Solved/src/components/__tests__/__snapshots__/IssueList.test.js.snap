// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueList should render 1`] = `
"<div class=\\"ui relaxed divided list\\">
  <div class=\\"item\\"><i class=\\"large github middle aligned icon\\"></i>
    <div class=\\"content\\"><a class=\\"header\\" href=\\"https://github.com/microsoft/vscode/issues/68\\" target=\\"_blank\\" rel=\\"noreferrer\\">Git: Support git history in VSCode</a>
      <div class=\\"description\\"></div>
    </div>
  </div>
  <div class=\\"item\\"><i class=\\"large github middle aligned icon\\"></i>
    <div class=\\"content\\"><a class=\\"header\\" href=\\"https://github.com/microsoft/vscode/issues/127\\" target=\\"_blank\\" rel=\\"noreferrer\\">Provide option to opt out of line ending normalisation for files</a>
      <div class=\\"description\\"></div>
    </div>
  </div>
</div>"
`;
